### PR TITLE
Add `identifier` and `description` annotations to operators

### PR DESCRIPTION
**Pull request by jaclarke**

### CODE REVIEW

File: edb/lib/cal.edgeql [hunk 1]
```diff
@@ -279,6 +279,9 @@ cal::date_get(dt: cal::local_date, el: std::str) -> std::float64
 
 CREATE INFIX OPERATOR
 std::`+` (l: std::datetime, r: cal::relative_duration) -> std::datetime {
+    CREATE ANNOTATION std::suggested_name := 'plus';
```
Comment (elprans) [inline edb/lib/cal.edgeql:282]: `std::identifier` perhaps?

**Comment by elprans** [comment]
The tests are failing on the last commit due to a compiler bug.  Please send it as a separate PR and let's land the annotations.

File: edb/lib/cal.edgeql [hunk 2]
```diff
@@ -324,13 +335,18 @@ std::`=` (l: cal::local_datetime, r: cal::local_datetime) -> std::bool {
 CREATE INFIX OPERATOR
 std::`?=` (l: OPTIONAL cal::local_datetime,
            r: OPTIONAL cal::local_datetime) -> std::bool {
+    CREATE ANNOTATION std::identifier := 'coaleq';
```
Comment (1st1) [inline edb/lib/cal.edgeql:338]: As I mentioned on the call let's put underscores in all compound names, e.g.:

```suggestion
    CREATE ANNOTATION std::identifier := 'coal_eq';
```

File: edb/lib/std/30-strfuncs.edgeql [hunk 3]
```diff
@@ -54,41 +64,57 @@ std::`?!=` (l: OPTIONAL std::str, r: OPTIONAL std::str) -> std::bool {
...
+    CREATE ANNOTATION std::description :=
+        'Case-sensitive simple string matching.';
     SET volatility := 'Immutable';
     USING SQL EXPRESSION;
 };
 
 
 CREATE INFIX OPERATOR
 std::`ILIKE` (string: std::str, pattern: std::str) -> std::bool {
+    CREATE ANNOTATION std::identifier := 'ilike';
```
Comment (1st1) [inline edb/lib/std/30-strfuncs.edgeql:86]: "ilike" should stay as is though (i.e. no underscore)

**Review by elprans — APPROVED** [review]

**Pull request merged**